rest/admin_api.go

```diff
@@ -601,20 +601,24 @@ func externalUserName(name string) string {
 	return name
 }
 
-func marshalPrincipal(princ auth.Principal) ([]byte, error) {
+func marshalPrincipal(princ auth.Principal, includeDynamicGrantInfo bool) ([]byte, error) {
 	name := externalUserName(princ.Name())
 	info := db.PrincipalConfig{
 		Name:             &name,
 		ExplicitChannels: princ.ExplicitChannels().AsSet(),
 	}
 	if user, ok := princ.(auth.User); ok {
-		info.Channels = user.InheritedChannels().AsSet()
 		info.Email = user.Email()
 		info.Disabled = user.Disabled()
 		info.ExplicitRoleNames = user.ExplicitRoles().AllChannels()
-		info.RoleNames = user.RoleNames().AllChannels()
+		if includeDynamicGrantInfo {
+			info.Channels = user.InheritedChannels().AsSet()
+			info.RoleNames = user.RoleNames().AllChannels()
```

Comment:
What do you think about refactoring timedSet.AllChannels() to timedSet.AllKeys() in this PR?  This isn't the first time I've spent a few minutes trying to work out why we're calling AllChannels() on a set of roles.

Reply:
Yeah that makes sense. To be honest reading this again it took me a sec to realize that wasn't getting the channels on a role...